param.go

```diff
@@ -146,36 +146,26 @@ func (pl paramList) Build(containerStore) (reflect.Value, error) {
 // to the underlying constructor.
 func (pl paramList) BuildList(c containerStore) ([]reflect.Value, error) {
 	args := make([]reflect.Value, len(pl.Params))
-	argsBuilt := make([]bool, len(pl.Params))
 	allContainers := c.getStoresFromRoot()
 	for i, p := range pl.Params {
-		var err error
-		var arg reflect.Value
 		// iterate through the tree path of scopes.
 		for _, c := range allContainers {
-			if arg, err = p.Build(c); err == nil {
+			arg, err := p.Build(c)
+			if err == nil {
 				args[i] = arg
```

Comment:
Do we need to `break` the allContainers loop here to stop instantiating the value in the children?

```
containerLoop:
for _, c := range allContainers {
  // ..
  if err == nil {
    args[i] = arg
    break containerLoop
```

Otherwise this will call the child's constructors for that type as well, right?

---

While we're at it, is this valid:

```
root := New()
root.Provide(func() Foo { ... })

child := root.Child("foo")
child.Provide(func() Foo { ... })
```

If so, does an `Invoke(func(Foo))` mean we want the child's version?
And if that's the case, the scope iteration order should probably be reversed?